packages/builders/src/components/ActionRow.ts

```diff
@@ -99,37 +70,120 @@ export class ActionRowBuilder<ComponentType extends AnyComponentBuilder> extends
 	 * ```
 	 */
 	public constructor({ components, ...data }: Partial<APIActionRowComponent<APIActionRowComponentTypes>> = {}) {
-		super({ type: ComponentType.ActionRow, ...data });
-		this.components = (components?.map((component) => createComponentBuilder(component)) ?? []) as ComponentType[];
+		super();
+		this.data = {
+			...structuredClone(data),
+			type: ComponentType.ActionRow,
+			components: components?.map((component) => createComponentBuilder(component)) ?? [],
```

Comment:
Consistency with `StringSelectMenu`
```suggestion
	public constructor({ components = [], ...data }: Partial<APIActionRowComponent<APIActionRowComponentTypes>> = {}) {
		super();
		this.data = {
			...structuredClone(data),
			type: ComponentType.ActionRow,
			components: components.map((component) => createComponentBuilder(component)),
```